snuba/query/data_source/join.py

```diff
@@ -20,6 +20,18 @@ class JoinModifier(Enum):
     SEMI = "SEMI"
 
 
+class JoinClass(Enum):
+    ONE_2_ZERO_OR_ONE = "1-0/1"
```

Comment:
we can skip all the relationships that have a possible 0 on the left side. They make little sense in a graph data model (if you do not have a node to navigate from the result cannot be a subgraph) even though we could technically run them in clickhouse.
So we can remove:
ONE_2_ZERO_OR_ONE
ZERO_OR_ONE_2_ZERO_OR_ONE
ZERO_OR_ONE_2_ONE
ZERO_OR_ONE_2_N
N_2_ZERO_OR_ONE

And have:
1/0-1
1/1
1/1-N
1/0-N
N/1
N0-1

I don't think we need to support n:m either as of now. It may come one day.

Reply:
I am confused, if we are not allowing 0/1 on the left side why would we keep `1/0 -> 1`?

Reply:
Ok I think I messed up the notation:
```
1 : 0..1
1 : 1
1 : 1..n
1 : 0..n
n : 1 
n : 0..1
```

there is no 0..1 : 1 allowed.